src/lib/interactionManager/interaction.ts

```diff
@@ -163,6 +165,50 @@ export class Interaction {
     )
   }
 
+  public async issueSelectedCredentials(
+    offerMap?: {
+      [k: string]: (inp?: any) => Promise<{ claim: any, metadata?: any, subject?: string }>
```

Comment:
Why is this map of promises and not a map of objects?

Reply:
Also why is it optional?

Reply:
Each handler returns a promise to allow for doing async things during the actual issuance process, in case the app logic requires it. This is so that we let this happen last second instead of requiring the app (backend) to already do all the work before even knowing what the "offer selection" is :D

Reply:
It is optional to allow for the "simple" use case of just empty claims